source/attribute/AttributeEffectType.h

```diff
@@ -0,0 +1,50 @@
+/* AttributeEffectType.h
+Copyright (c) 2023 by tibetiroka
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef ATTRIBUTE_EFFECT_TYPE_H_
+#define ATTRIBUTE_EFFECT_TYPE_H_
+
+// The different effects of an attribute. Each one has multiple variants,
+// each offset by ATTRIBUTE_EFFECT_COUNT. The variants are:
+// base (not offset), multiplier, relative, and relative multiplier.
+enum AttributeEffectType : int {
+	SHIELDS,
+	HULL,
+	THRUST,
+	REVERSE_THRUST,
+	TURN,
+	ACTIVE_COOLING,
+	RAMSCOOP,
+	CLOAK,
+	COOLING,
+	FORCE,
+	ENERGY,
+	FUEL,
+	HEAT,
+	DISCHARGE,
+	CORROSION,
```

Comment:
`corrision` is `hull over time`
`burn` is `heat over time`
`leak` is `fuel over time`
`ion` is `energy over time`

So I would actually expect those `over time` effectTypes to be identified by their base-type, and have a variant for those (similar to the base, multiplier, relative and relative multiplier variants).

Reply:
note that this is only for the deep in-engine support. I do feel that we should keep them as separate keywords in the syntax as proposed for #9025. (So the deep engine-support and the keywords would differ at this point.)

Reply:
That sounds doable, though parsing would be slightly more complicated because of the different keywords. I'll definitely look into this.